app/services/ts-importer.ts

```diff
@@ -0,0 +1,421 @@
+
+interface ITSCompositorLayer {
+  id: string;
+  name: string;
+  position: {
+    bottom: number;
+    left: number;
+    right: number;
+    top: number;
+  };
+  plugin: ITSLayerContent;
+}
+
+interface ITSLayerContent {
+  id: TTSLayerContent;
+}
+
+type TTSPlugin =
+  | 'nativeMedia'
+  | 'compositor'
+  | 'windowsVideoCapture'
+  | 'nativeText'
+  | 'browserSource'
+  | 'colorInput'
+  | 'primaryScreenShare';
+type TTSLayerContent = 'windowsVideoCapture';
+
+export class TwitchStudioImporterService extends Service {
+  @Inject() sceneCollectionsService: SceneCollectionsService;
+  @Inject() scenesService: ScenesService;
+  @Inject() settingsService: SettingsService;
+  @Inject() sourcesService: SourcesService;
+  @Inject() defaultHardwareService: DefaultHardwareService;
+  @Inject() videoService: VideoService;
+
+  @RunInLoadingMode()
+  async import() {
+    await this.load();
+  }
+
+  async load() {
+    if (!this.isTwitchStudioIntalled()) {
+      console.error('Twitch Studio is not installed!');
+      return;
+    }
+
+    const config = JSON.parse(fs.readFileSync(this.layoutsFile).toString()) as ITSConfig;
+
+    await this.sceneCollectionsService.create({
+      name: 'Twitch Studio Imported',
+      setupFunction: async () => {
+        this.setupVideo(config);
+        this.importScenes(config);
+
+        return this.scenesService.views.scenes.length !== 0;
+      },
+    });
+  }
+
+  setupVideo(config: ITSConfig) {
+    this.settingsService.setSettingValue(
+      'Video',
+      'Base',
+      `${config.graphics.canvasWidth}x${config.graphics.canvasHeight}`,
+    );
+  }
+
+  async importScenes(config: ITSConfig) {
+    for (const layout of config.layoutSettings.layouts) {
+      // Skip virtualcam scene
+      if (layout.collections[0]?.name === 'unlisted') continue;
+
+      const scene = this.scenesService.createScene(layout.displayName, { makeActive: true });
+
+      // Look for a corresponding compositor
+      const compositor = config.graph.nodes.find(n => n.id === layout.videoOutput);
+
+      if (compositor.pluginId === 'compositor') {
+        // Load each layer from the compositor
+        for (const input of compositor.inputs) {
+          await this.importSource(
+            compositor.pluginSettings.layers[input],
+            config.graph.nodes.find(n => n.id === input),
+          );
+        }
+      } else {
+        console.error(
+          `Expected layout videoOutput to be a compositor but instead is ${compositor.pluginId}`,
+        );
+      }
+    }
+  }
+
+  async importSource(layer: ITSCompositorLayer, node: TTSNode) {
+    let item: SceneItem;
+    let noCrop = false;
+    let widthOverride: number;
+    let heightOverride: number;
+
+    if (node.pluginId === 'windowsVideoCapture') {
+      let webcamSource = this.sourcesService.views.getSourcesByType('dshow_input')[0];
+
+      if (!webcamSource) {
+        webcamSource = this.sourcesService.createSource(layer.name, 'dshow_input');
```

Comment:
Don't we have a specialized method for creating these that's supposedly abstracting away the differences between Windows and MacOS? I might be thinking of a different source type, but I remember seeing this somewhere.

Reply:
It's not wrapped in a method, but we do have that code somewhere.  This importer is windows only though.